File: benchmarks/suite/alloc.cpp
#include <cstdio>
struct Node { int x, y; };
int main() {
    int N = 1000000, sink = 0;
    for (int i = 0; i < N; i++) {
        Node *n = new Node{i, i+1};
        sink += n->x;
        delete n;
    }
    printf("%d\n", sink);
}

File: benchmarks/suite/alloc.js
class Node {
    constructor(a, b) { this.x = a; this.y = b; }
}
const N = 1_000_000;
let sink = 0;
for (let i = 0; i < N; i++) {
    const n = new Node(i, i + 1);
    sink += n.x;
}
console.log(sink);
